### ISSUE TITLE
Data race with stack traces

### ISSUE THREAD

**Issue by fxfactorial** [libp2p/go-libp2p-autonat#79]
```
==================
==================
WARNING: DATA RACE
Write at 0x00c0000a4b68 by goroutine 1876:
  github.com/libp2p/go-libp2p-autonat.(*AmbientAutoNAT).probeNextPeer()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-autonat@v0.2.2/autonat.go:352 +0x34f

Previous read at 0x00c0000a4b68 by goroutine 130:
  github.com/libp2p/go-libp2p-autonat.(*AmbientAutoNAT).scheduleProbe()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-autonat@v0.2.2/autonat.go:223 +0xe5
  github.com/libp2p/go-libp2p-autonat.(*AmbientAutoNAT).background()
      /home/edgar/go/pkg/mod/github.com/libp2p/go-libp2p-autonat@v0.2.2/autonat.go:207 +0x212

```

**Comment by bonedaddy** [comment]
I've also just encountered this:


```
WARNING: DATA RACE
Read at 0x00c00015c5e8 by goroutine 42:
  github.com/libp2p/go-libp2p-autonat.(*AmbientAutoNAT).scheduleProbe()
      /home/travis/gopath/pkg/mod/github.com/libp2p/go-libp2p-autonat@v0.2.2/autonat.go:223 +0xe5
  github.com/libp2p/go-libp2p-autonat.(*AmbientAutoNAT).background()
      /home/travis/gopath/pkg/mod/github.com/libp2p/go-libp2p-autonat@v0.2.2/autonat.go:207 +0x212
Previous write at 0x00c00015c5e8 by goroutine 100:
  github.com/libp2p/go-libp2p-autonat.(*AmbientAutoNAT).probeNextPeer()
      /home/travis/gopath/pkg/mod/github.com/libp2p/go-libp2p-autonat@v0.2.2/autonat.go:352 +0x372
Goroutine 42 (running) created at:
  github.com/libp2p/go-libp2p-autonat.New()
      /home/travis/gopath/pkg/mod/github.com/libp2p/go-libp2p-autonat@v0.2.2/autonat.go:120 +0x935
  github.com/libp2p/go-libp2p/config.(*Config).NewNode()
      /home/travis/gopath/pkg/mod/github.com/libp2p/go-libp2p@v0.8.1/config/config.go:343 +0x61f
  github.com/libp2p/go-libp2p.NewWithoutDefaults()
      /home/travis/gopath/pkg/mod/github.com/libp2p/go-libp2p@v0.8.1/libp2p.go:71 +0x130
  github.com/RTradeLtd/libp2p-testutils.NewLibp2pHostAndDHT()
      /home/travis/gopath/pkg/mod/github.com/libp2p/go-libp2p@v0.8.1/libp2p.go:57 +0x433
  github.com/RTradeLtd/node/v3/bootstrap.Test_Bootstrap()
      /home/travis/gopath/src/github.com/RTradeLtd/node/bootstrap/bootstrap_test.go:29 +0x4b2
  testing.tRunner()
      /home/travis/.gimme/versions/go1.13.10.linux.amd64/src/testing/testing.go:909 +0x199
```

Go version: 1.13.10
Autonat Version: v0.2.2

**Comment by willscott** [comment]
This should be fixed in v0.2.3

**Comment by willscott** [comment]
closing as obsolete
